### PR TITLE
Update gfxrecon-convert's README.md

### LINKED ISSUE #1359: Convert README.md is out of Date
It is missing features and shows an older "schema" on Vulkan side.

It completely ignores D3D12.

See: https://github.com/LunarG/gfxreconstruct/blob/dev/tools/convert/README.md

### PR DESCRIPTION
**Pull request by nickdriscoll-lunarg**
Fixes #1359 

Issue #1359 pointed out that gfxrecon-convert's documentation was missing features, was outdated in the schema it described and showed in examples, and failed to mention D3D12 at all.

This PR updates the README.md such that it covers the missing features, the examples and explanations of the output schema are consistent with the tool's current behavior, and D3D12 is mentioned with examples when appropriate.

### CODE REVIEW

File: tools/convert/README.md [hunk 1]
```diff
@@ -24,38 +27,48 @@ Optional arguments:
...
+With `"--format jsonl"`, because each JSON object is on its own line, line oriented tools such as
 grep, sed, head, and split can be applied ahead of JSON-aware ones which
 are heavier-weight to reduce their workload on large captures.
 
 
 ## JSON Structure
 
-The tool's output is an ordered list of strings, one per line, each line a valid
-JSON document (JSON Lines). Below are the first few lines from a capture of vkcube, truncated to 200 columns.
+The tool's output is an ordered list of JSON structures, one per line in `--format jsonl` mode, each line a valid
```
Comment (bradgrantham-lunarg) [inline tools/convert/README.md:58]: Could you clarify this a hair?  Maybe separate out the tool's output in default `json` mode versus `jsonl` mode; maybe something like:
```
The tool's output in default `json` mode is an ordered list of JSON structures, or a single JSON structure per line in `--format jsonl` mode.  Each structure represents metadata, a metadata command, or a graphics API function or method call.  Below are the first few lines from a capture of vkcube with convert applied with `--format jsonl`, truncated to 200 columns.
```

Comment (bradgrantham-lunarg) [reply tools/convert/README.md:58]: Are all the subsequent examples either an extract of one structure from the JSON list or a pretty-print of a line from JSONL?  Could you add that as a note?

File: tools/convert/README.md [hunk 2]
```diff
@@ -555,19 +705,19 @@ gfxrecon-convert --output stdout vkcube.f1.gfxr | egrep -v "^{\"header\":{\"" |
 This can be stripped-down to just the values:
 
 ```bash
-gfxrecon-convert --output stdout vkcube.f1.gfxr | egrep -v "^{\"header\":{\"" | jq -c '.vkFunc.args = (.vkFunc.args | to_entries | map_values(.value))'
+gfxrecon-convert --output stdout vkcube.f1.gfxr | egrep -v "^{\"header\":{\"" | jq -c '.function.args = (.function.args | to_entries | map_values(.value))'
```
Comment (bradgrantham-lunarg) [inline tools/convert/README.md:708]: I think this example and other examples may need to be updated to add `--format jsonl` when appropriate

File: tools/convert/README.md [hunk 3]
```diff
@@ -579,7 +729,7 @@ These might be useful to keep beside a set of multi-gigabyte binary traces to
 allow fast grepping when looking for traces that use particular named arguments.
 
 ```bash
-gfxrecon-convert --output stdout vkcube.f1.gfxr | jq  -c "[.. | objects | keys[]] | unique" | sed "s/\"args\",//" | sed "s/\"index\",//" | sed "s/\"name\",//" | sed "s/\"return\",//" | sed "s/,\"vkFunc\"//" | sort | uniq
+gfxrecon-convert --output stdout vkcube.f1.gfxr | jq  -c "[.. | objects | keys[]] | unique" | sed "s/\"args\",//" | sed "s/\"index\",//" | sed "s/\"name\",//" | sed "s/\"return\",//" | sed "s/,\"function\"//" | sort | uniq
```
Comment (bradgrantham-lunarg) [inline tools/convert/README.md:732]: I think this example needs `--format jsonl`.  I didn't check any further; could you verify whether the examples need `--format jsonl` or not?

**Review by bradgrantham-lunarg — CHANGES_REQUESTED** [review]

**Comment by nickdriscoll-lunarg** [comment]
Clarified line 58 and added "--format jsonl" to the examples that needed it.

**Review by bradgrantham-lunarg — APPROVED** [review]

**Pull request merged**